File: Java/Linked List/Singly Circular Linked List/SinglyCL10.java
/*
=====================================================================

Name            : Singly Circular Linked List - Function Declarations
Program Number  : 010
Description     : This program demonstrates the basic framework of
                  a Singly Circular Linked List class.

                  The class contains declarations of all major
                  linked list operations such as insertion,
                  deletion, display, and count. Function bodies
                  are kept empty and will be implemented in
                  subsequent programs.

Concepts Used   : Singly Circular Linked List,
                  Class Design,
                  Method Declaration,
                  Constructor,
                  Object-Oriented Programming

Input           : None

Output          : Object creation of Singly Circular Linked List

Date            : July 2026
Language        : Java
Author          : Atharv Tushar Bhosale

=====================================================================
*/

class node
{
    public int data;
    public node next;

    node(int no)
    {
        this.data = no;
        this.next = null;
    }
}

class SinglyCL
{
    public node first;
    public node last;
    public int iCount;

    public SinglyCL()
    {
        this.first = null;
        this.last = null;
        this.iCount = 0;
    }

    public void Disply()
    {}

    public int Count()
    {
        return iCount;
    }

    public void InsertFirst(int no)
    {}

    public void InsertLast(int no)
    {}

    public void InsertAtPos(int no, int iPos)
    {}

    public void DeleteFirst()
    {}

    public void DeleteLast()
    {}

    public void DeleteAtPos(int iPos)
    {}
}

class SinglyCL10
{
    public static void main(String A[])
    {
        SinglyCL sobj = new SinglyCL();
    }
}

/*
=====================================================================

Sample Output :

(No Output)

=====================================================================

Observation :

- Singly Circular Linked List class is created.
- Constructor initializes linked list members.
- Display, Count, Insert and Delete methods are declared.
- Count method returns current node count.
- Method implementations will be added later.

=====================================================================
*/

File: C++/Linked List/Singly Circular Linked List/SinglyCL10.cpp
/*
=====================================================================

Name            : Singly Circular Linked List - InsertFirst Operation
Program Number  : 010

Description     : This program implements the InsertFirst()
                  operation for a Singly Circular Linked List.

                  A new node is created and inserted at the
                  beginning of the linked list.

                  If the list is empty, both first and last
                  pointers point to the new node.

                  After insertion, the circular connection is
                  maintained by connecting the last node to
                  the first node.

Concepts Used   : Singly Circular Linked List,
                  Dynamic Memory Allocation,
                  InsertFirst Operation,
                  Circular Linking

Input           : Node data

Output          : Nodes inserted successfully

Date            : July 2026
Language        : C++
Author          : Atharv Tushar Bhosale

=====================================================================
*/

#include<iostream>
using namespace std;

class node
{
    public:
        int data;
        node* next;

        node(int no)
        {
            data = no;
            next = NULL;
        }
};

class SinglyCL
{
    public:
        node* first;
        node* last;
        int iCount;

        SinglyCL()
        {
            first = NULL;
            last = NULL;
            iCount = 0;
        }

        void InsertFirst(int no)
        {
            node* newn = new node(no);

            if(first == NULL && last == NULL)
            {
                first = newn;
                last = newn;
            }
            else
            {
                newn->next = first;
                first = newn;
            }

            last->next = first;
            iCount++;
        }
};

int main()
{
    SinglyCL sobj;

    sobj.InsertFirst(51);
    sobj.InsertFirst(21);
    sobj.InsertFirst(11);

    cout<<"Nodes inserted successfully"<<endl;

    return 0;
}

/*
=====================================================================

Output :

Nodes inserted successfully

=====================================================================
*/
